### PR TITLE
bump runc to v1.2.0

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

https://github.com/opencontainers/runc/releases/tag/v1.2.0

#### Which issue(s) this PR fixes:

xref https://github.com/kubernetes/enhancements/issues/4205

PSI is pending on this due to https://github.com/containerd/cgroups/pull/308 (https://github.com/containerd/containerd/pull/9583) Containerd 2.0 only(not included in v1.7 containerd)

#### Does this PR introduce a user-facing change?
```release-note
Bump runc to v1.2.0
```

### CODE REVIEW

**Comment by pacoxu** [comment]
`check-dependency-stats`
```
-Direct Dependencies: 188 
-Transitive Dependencies: 333 
-Total Dependencies: 371 
+Direct Dependencies: 189 
+Transitive Dependencies: 334 
+Total Dependencies: 372 

... 
 github.com/checkpoint-restore/go-criu/v5
+github.com/checkpoint-restore/go-criu/v6
-github.com/frankban/quicktest
+github.com/moby/sys/user
+github.com/moby/sys/userns
-github.com/shurcooL/sanitized_anchor_name
```

File: hack/unwanted-dependencies.json [hunk 1]
```diff
@@ -216,8 +216,8 @@
       ],
       "golang.org/x/exp": [
         "github.com/antlr4-go/antlr/v4",
-        "github.com/cilium/ebpf",
         "github.com/google/cel-go",
+        "github.com/opencontainers/runc",
```
Comment (pacoxu) [inline hack/unwanted-dependencies.json:220]: Adding this
```
github.com/opencontainers/runc@v1.2.0 golang.org/x/exp@v0.0.0-20230224173230-c95f2b4c22f2
```

Before
```
github.com/opencontainers/runc@v1.1.15 github.com/cilium/ebpf@v0.7.0
github.com/cilium/ebpf@v0.11.0 golang.org/x/exp@v0.0.0-20230224173230-c95f2b4c22f2
```

**Comment by dims** [comment]
@pacoxu we should not do this yet... we try to keep the runc binary and runc library in sync as much as possible.

**Comment by pacoxu** [comment]
link @dims comments https://github.com/google/cadvisor/pull/3083#issuecomment-2429105936

**Comment by pacoxu** [comment]
if so, I am afraid that the PSI KEP should not be target to v1.32.

File: hack/unwanted-dependencies.json (hunk 1 above)
Comment (liggitt) [reply hack/unwanted-dependencies.json:220]: aww, I was excited we were dropping ebpf ... can we figure out why runc is using exp and open a PR to help them drop usage?

**Review by liggitt — COMMENTED** [review]
one (non-blocking) question about helping clean up x/exp usage in runc, lgtm otherwise

/approve
for dependency changes

/hold for node approver ack of the runc bump

**Comment by pacoxu** [comment]
According to https://github.com/kubernetes/kubernetes/issues/128157 and https://groups.google.com/g/kubernetes-sig-node/c/ce2yLXg5SDw, we have a plan to migrate, but I am not sure the timeline of the todo items.

File: hack/unwanted-dependencies.json (hunk 1 above)
Comment (pacoxu) [reply hack/unwanted-dependencies.json:220]: I opened https://github.com/opencontainers/runc/issues/4469.

Comment (pacoxu) [reply hack/unwanted-dependencies.json:220]: This seems to be a bug of `go mod graph`.

https://github.com/opencontainers/runc/issues/4469#issuecomment-2431238789

```
(base) ➜  runc git:(main) go mod graph                     
github.com/opencontainers/runc golang.org/x/exp@v0.0.0-20230224173230-c95f2b4c22f2
github.com/cilium/ebpf@v0.16.0 golang.org/x/exp@v0.0.0-20230224173230-c95f2b4c22f2
```

Runc does not use `golang.org/x/exp` directly. As we don't use ` github.com/cilium/ebpf`, `golang.org/x/exp` should be excluded as well.

**Comment by dims** [comment]
/close

in favor of https://github.com/kubernetes/kubernetes/pull/128276

File: hack/unwanted-dependencies.json (hunk 1 above)
Comment (kolyshkin) [reply hack/unwanted-dependencies.json:220]: As I mentioned in https://github.com/kubernetes/kubernetes/pull/128276#issue-2606584699, there seems to be a bug.

Comment (liggitt) [reply hack/unwanted-dependencies.json:220]: > As we don't use  github.com/cilium/ebpf

It doesn't matter, ebpf is in your dependency tree, so you inherit the dependencies of ebpf.

```
runc]$ git status
HEAD detached at v1.2.0
nothing to commit, working tree clean

runc]$ go mod graph | grep -E 'exp|ebpf'
github.com/opencontainers/runc github.com/cilium/ebpf@v0.16.0
github.com/opencontainers/runc golang.org/x/exp@v0.0.0-20230224173230-c95f2b4c22f2
...
github.com/cilium/ebpf@v0.16.0 golang.org/x/exp@v0.0.0-20230224173230-c95f2b4c22f2
```

and pass those on to your dependencies

**Pull request closed without merging**